src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -6110,10 +6110,11 @@ private SqlAuthenticationToken getFedAuthToken(SqlFedAuthInfo fedAuthInfo) throw
         }
 
         while (true) {
+            int millisecondsRemaining = timerRemaining(timerExpire);
```

Comment:
If this is 0 or negative does it make sense to try the calls below but not throw timed out exception here (or from timerRemaining call)? Or just let the first calls throw the exception?

Reply:
there's min value of 1ms, but the question is still valid.

Reply:
right, we are just letting the first call throw it